packages/shell-api/src/decorators.ts

```diff
@@ -36,27 +36,81 @@ export interface ShellResultSourceInformation {
 }
 
 export interface ShellResult {
-  value: any;
-  type: string;
+  /// The original result of the evaluation, as it would be stored e.g. as a
+  /// variable inside the shell.
+  rawValue: any;
+
+  /// A version of the raw value that is usable for printing, e.g. what the
+  /// shell would print.
+  printable: any;
+
+  /// The type of the shell result. This refers to built-in shell types, e.g.
+  /// `Cursor`; all unknown object types and primitives are given the
+  /// type `null`.
+  type: string | null;
+
+  /// Optional information about the original data source of the result.
   source?: ShellResultSourceInformation;
+
+  /// @deprecated Use either `printable` or `rawValue`.
+  value: any;
 }
 
 export class ShellApiClass implements ShellApiInterface {
   help: any;
-  [asShellResult](): any {
+  get [shellApiType](): string {
     throw new MongoshInternalError('Shell API Type did not use decorators');
   }
-  _asPrintable(): any {
+  set [shellApiType](value: string) {
+    addHiddenDataProperty(this, shellApiType, value);
+  }
+  [asPrintable](): any {
     return Object.assign({}, this);
   }
+
+  /// @deprecated: Use `toShellResult(value).printable` instead.
```

Comment:
I'd just drop it and fix it in the consumers, not even sure anyone is using it right now, didn't Jetbrains switch to use the `ShellEvaluator`?

Reply:
They are still calling `_asPrintable()` in a few places right now … I can try to figure out how to move them away from it, but for me, the `java-shell` test suite is not passing locally on `master` to begin with